web/src/components/dashboard/SidebarCardFundingNudge.tsx

```diff
@@ -13,16 +23,104 @@ type Props = {
 
 export const SidebarCardFundingNudge = (props: Props): ReactElement => {
   const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [profileData, setProfileData] = useState<ProfileData>(createEmptyProfileData());
+
+  const { queueUpdateTaskProgress } = useUpdateTaskProgress();
+  const close = (): void => {};
+  const onSuccess = (): void => {};
   const router = useRouter();
   const { updateQueue, business } = useUserData();
 
-  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+  useMountEffectWhenDefined(() => {
     if (!business) return;
+    setProfileData(business.profileData);
+  }, business);
+
+  const displayBusinessName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "businessName"
+    );
+  };
+
+  const displayResponsibleOwnerName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "responsibleOwnerName"
+    );
+  };
+
+  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+    if (!business || !updateQueue) return;
     await updateQueue
       ?.queueProfileData({
         operatingPhase: OperatingPhaseId.UP_AND_RUNNING,
       })
       .update();
+
+    const encryptedTaxId =
+      profileData.taxId === business.profileData.taxId ? profileData.encryptedTaxId : undefined;
+
+    try {
+      let businessNameToSubmitToTaxApi = "";
+
+      if (displayBusinessName()) {
+        businessNameToSubmitToTaxApi = profileData.businessName;
+      }
+      if (displayResponsibleOwnerName()) {
+        businessNameToSubmitToTaxApi = profileData.responsibleOwnerName;
```

Comment:
dust - This is quite nitpicky and you can totally ignore this. 

I would have structured this a bit differently personally. 

getBusinessNameToSubmitToTaxApi to hold both the getting logic for displayBusinessName and displayResponsibleOwnerName as well as choosing which string to return.

Reply:
Oh, it also looks like this logic is repeated below in taxAccess so it would probably be good to throw this in a helper file and reuse it.

Reply:
Attempts at refactoring this (i.e. -- moving this logic to a `lib` folder to reuse in `SidebarCardFundingNudge.tsx`) or to put into a switch function (or similar) caused the jest tests to fail.